src/test/java/com/cdancy/bitbucket/rest/features/InsightsApiMockTest.java

```diff
@@ -0,0 +1,478 @@
+import com.cdancy.bitbucket.rest.domain.common.RequestStatus;
+import com.cdancy.bitbucket.rest.domain.insights.Annotation;
+import com.cdancy.bitbucket.rest.domain.insights.AnnotationsResponse;
+import com.cdancy.bitbucket.rest.domain.insights.InsightReport;
+import com.cdancy.bitbucket.rest.domain.insights.InsightReportData;
+import com.cdancy.bitbucket.rest.domain.insights.InsightReportPage;
+import com.cdancy.bitbucket.rest.options.CreateAnnotations;
+import com.cdancy.bitbucket.rest.options.CreateInsightReport;
+import com.google.common.collect.ImmutableMap;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(groups = "unit", testName = "InsightApiMockTest")
+public class InsightsApiMockTest extends BaseBitbucketMockTest {
+    private final String projectKey = "PRJ";
+    private final String repoKey = "my-repo";
+    private final String commitHash = "abcdef0123abcdef4567abcdef8987abcdef6543";
+
+    private final String getMethod = "GET";
+    private final String deleteMethod = "DELETE";
+    private final String putMethod = "PUT";
+
+    private final String restApiPath = "/rest/insights/";
+
+    private final String projectsKeyword = "/projects/";
+    private final String repoKeyword = "/repos/";
+    private final String commitKeyword = "/commits/";
+    private final String annotationsKeyword = "/annotations";
+    private final String reportsKeyword = "/reports";
+
+
+    private final String qwertyKeyword = "qwerty";
+    private final String limitKeyword = "limit";
+    private final String startKeyword = "start";
+    private final String annotationsJsonFile = "/annotations.json";
+    private final String mockPath = ".gitignore";
+
+    public void testListReports() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/insight-report-page.json")).setResponseCode(200));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final InsightReportPage insightReportPage = api.listReports(projectKey, repoKey, commitHash, 100, 0);
+            assertThat(insightReportPage).isNotNull();
+            assertThat(insightReportPage.values()).isNotEmpty();
+            assertThat(insightReportPage.errors()).isEmpty();
+
+            final Map<String, ?> queryParams = ImmutableMap.of(limitKeyword, 100, startKeyword, 0);
+
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword, queryParams);
```

Comment:
:warning: Line is longer than 140 characters (found 142). (#QR119:84#)-CHECKSTYLE_LineLengthCheck